sql/core/src/test/scala/org/apache/spark/sql/DataFrameAggregateSuite.scala

```diff
@@ -1091,6 +1091,25 @@ class DataFrameAggregateSuite extends QueryTest
     val df = spark.sql(query)
     checkAnswer(df, Row(0, "0", 0, 0) :: Row(-1, "1", 1, 1) :: Row(-2, "2", 2, 2) :: Nil)
   }
+
+  test("SPARK-34713: group by CreateStruct with ExtractValue") {
```

Comment:
Thank you for adding this test case. BTW, do you think we can have a narrow-downed test case in `catalyst` module?

Reply:
We don't have a UT for `ResolveCreateNamedStruct` yet. The bug is trivial so I didn't spend time building a new UT.